zebra/dplane_fpm_nl.c

```diff
@@ -1467,6 +1467,10 @@ static int fpm_nl_process(struct zebra_dplane_provider *prov)
 		thread_add_timer(fnc->fthread->master, fpm_process_queue,
 				 fnc, 0, &fnc->t_dequeue);
 
+	/* Ensure dataplane thread is rescheduled if we hit the work limit */
+	if (counter >= limit)
```

Comment:
wouldn't this be better having a test to see if we actually have data to process instead of just blindly doing it?

Reply:
Basically the same reasoning as below - can be changed but will require a new accessor API as we cannot dereference `prov` here.